src_c/rect_impl.h

```diff
@@ -571,102 +571,132 @@ int RectOptional_Freelist_Num = -1;
 static InnerRect *
 RectExport_RectFromObject(PyObject *obj, InnerRect *temp)
 {
-    PrimitiveType val;
     Py_ssize_t length;
 
     if (RectCheck(obj)) {
         return &((RectObject *)obj)->r;
     }
-    if (PySequence_Check(obj) && (length = PySequence_Length(obj)) > 0) {
+
+    if (pgSequenceFast_Check(obj)) {
+        length = PySequence_Fast_GET_SIZE(obj);
+        PyObject **items = PySequence_Fast_ITEMS(obj);
+
         if (length == 4) {
-            if (!primitiveFromObjIndex(obj, 0, &val)) {
-                return NULL;
-            }
-            temp->x = val;
-            if (!primitiveFromObjIndex(obj, 1, &val)) {
+            if (!PrimitiveFromObj(items[0], &temp->x) ||
+                !PrimitiveFromObj(items[1], &temp->y) ||
+                !PrimitiveFromObj(items[2], &temp->w) ||
+                !PrimitiveFromObj(items[3], &temp->h)) {
                 return NULL;
             }
-            temp->y = val;
-            if (!primitiveFromObjIndex(obj, 2, &val)) {
-                return NULL;
-            }
-            temp->w = val;
-            if (!primitiveFromObjIndex(obj, 3, &val)) {
+            return temp;
+        }
+        else if (length == 2) {
+            if (!twoPrimitivesFromObj(items[0], &temp->x, &temp->y) ||
+                !twoPrimitivesFromObj(items[1], &temp->w, &temp->h)) {
                 return NULL;
             }
-            temp->h = val;
             return temp;
         }
-        if (length == 2) {
-            PyObject *sub = PySequence_GetItem(obj, 0);
-            if (!sub || !PySequence_Check(sub) ||
-                PySequence_Length(sub) != 2) {
-                PyErr_Clear();
-                Py_XDECREF(sub);
+        else if (length == 1) {
```

Comment:
```suggestion
        else if (PyTuple_Check(obj) && length == 1) {
```
For reasons mentioned in the other review comment